packages/react-reconciler/src/ReactFiber.js

```diff
@@ -204,6 +207,8 @@ function FiberNode(
 
   this.alternate = null;
 
+  this.snapshot = null;
```

Comment:
We should be able to implement this without adding a new field to every fiber.

Could we try storing it on the `stateNode` (instance) instead? It's confined to the commit phase, so should be fine.

Reply:
Yeah, there's a TODO about this 😄 This PR isn't ready yet.

Initially I was thinking maybe the `updateQueue` could be a good place to store the snapshot but I'm not sure if this would work out. I think `updateQueue` could be null when I'd need it in some cases.

I dunno about using the instance. Seems like we could have potential collision issues if we tried that?

Reply:
The snapshot is only used within a single commit phase so it shouldn't collide with anything

Reply:
I meant naming collisions. Writing to the instance might clobber an existing property.

Reply:
We already write to `__reactInternalMemoizedMaskedChildContext` and `__reactInternalMemoizedMergedChildContext` for legacy context. I think anything `__reactInternal` is fair game.